### PR TITLE
Add link to npm release details to the release docs

### PR DESCRIPTION
**Pull request by glendaviesnz**
## What?
Adds link to the npm release info anchor to reference at top of docs

## Why?
The npm details are a long way down the page, so saves people new to release process having to scroll through full doc to find the details.

## How?
Added a markdown anchor link

## Testing Instructions
View edited file and make sure link to NPM publishing details at top of page works as expected

### CODE REVIEW

**Review by gziolo — APPROVED** [review]
Thank you. We also need to include the note that the successful RC1 release triggers npm publishing job that need to be approved by the Gutenberg core team member. It's similar process with the approval for every stable plugin release that use svn secret tokens.

**Comment by glendaviesnz** [comment]
> We also need to include the note that the successful RC1 release triggers npm publishing job that need to be approved by the Gutenberg core team member

How [does this look](https://github.com/WordPress/gutenberg/pull/40867/files#diff-17d1ffea91f685c145076051d0b46384b2cf600635cc9429d877e168914aca1bR207)?

**Comment by glendaviesnz** [comment]
> We also need to include the note that the successful RC1 release triggers npm publishing job that need to be approved by the Gutenberg core team member

How [does this look](https://github.com/WordPress/gutenberg/pull/40867/files#diff-17d1ffea91f685c145076051d0b46384b2cf600635cc9429d877e168914aca1bR207) @gziolo?

**Comment by gziolo** [comment]
That's great @glendaviesnz! I added some clarification to the updated paragraph in https://github.com/WordPress/gutenberg/pull/40867/commits/f004bc097fc56f9fd6371acd55201da499d43ed9. We are storing npm token for the special https://www.npmjs.com/~gutenbergplugin account so the job can be approved by any @WordPress/gutenberg-core member. They don't need to have access to WordPress org on npm. That was one of the goals of the automation introduced to avoid maintaining a long list of maintainers on npm.

I also included the note similar one in [Publishing the release](https://github.com/WordPress/gutenberg/blob/f004bc097fc56f9fd6371acd55201da499d43ed9/docs/contributors/code/release.md#publishing-the-release) section.

File: docs/contributors/code/release.md [hunk 1]
```diff
@@ -204,7 +204,7 @@ There is also an option to perform [Standalone Bugfix Package Releases](#standal
 
 ### Synchronizing Gutenberg Plugin
 
-For each Gutenberg plugin release, we also publish to npm an updated version of WordPress packages. This is automated with the [Release Tool](#release-tool) that handles releases for the Gutenberg plugin.
+For each Gutenberg plugin release, we also publish to npm an updated version of WordPress packages. This is automated with the [Release Tool](#release-tool) that handles releases for the Gutenberg plugin. A successful RC1 release triggers the npm publishing job, and this needs to be approved by the Gutenberg Core team member. Locate the ["Build Gutenberg Plugin Zip" workflow](https://github.com/WordPress/gutenberg/actions/workflows/build-plugin-zip.yml) for the new version, and have it [approved](https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments#approving-or-rejecting-a-job).
```
Comment (ntsekouras) [inline docs/contributors/code/release.md]:
```suggestion
For each Gutenberg plugin release, we also publish to npm an updated version of WordPress packages. This is automated with the [Release Tool](#release-tool) that handles releases for the Gutenberg plugin. A successful RC1 release triggers the npm publishing job, and this needs to be approved by a Gutenberg Core team member. Locate the ["Build Gutenberg Plugin Zip" workflow](https://github.com/WordPress/gutenberg/actions/workflows/build-plugin-zip.yml) for the new version, and have it [approved](https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments#approving-or-rejecting-a-job).
```

**Pull request merged**